src/context/mapStateSlice.ts

```diff
@@ -33,13 +33,21 @@ const initialState: MapState = {
   layersData: [],
 };
 
+function keepLayer(layer: LayerType, payload: LayerType) {
+  // Simple function to control which layers can overlap.
+  if (payload.type === layer.type) {
+    return false;
+  }
+  return layer.id !== payload.id;
```

Comment:
This is true by definition if the first condition is met (if we reach this line, it implies `payload.type !== layer.type`, which by definition means that `id` isn't equal, unless we have duplicate ids, which isn't allowed). I'd just simplify this logic to `return payload.type !== layer.type` (and probably put it inline for now, until we add more complicated logic.